src/components/App.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useState } from 'react';
+import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
+
+import { IRecipe } from '../models/IRecipe';
+import { AddEditRecipe } from './AddEditRecipe';
+// import { ViewRecipe } from './ViewRecipe';
+import { BrowseRecipes } from './BrowseRecipes';
+
+export interface AppProps {
+  recipe: IRecipe;
+  recipes: IRecipe[];
```

Comment:
You want to make sure you don't duplicate state or props - will recipe also be in recipes? If you just want to track an active recipe you can use a prop like activeRecipeId.

Our props and state are kind of messy right now because we don't have the full picture of how they're going to be used (I think we'll have a much better picture once we start persisting state and using something like React router). We should try to keep it as clean and minimal as possible in the meantime, and make sure we do some serious cleaning up once the shape of the app is more clear.

Reply:
Good catch! Good news is with adding recipes, I don't even use recipe anymore.

Reply:
Yeah, I agree. I think we will fix it up when we are refactoring and have a better sense of how the app will look. This is just sometimes how it goes when learning something new and starting a new project